xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/Inputstream.h

```diff
@@ -0,0 +1,585 @@
+    unsigned int m_BlockAlign;
+  } INPUTSTREAM_INFO;
+
+  /*!
+   * @brief Structure to transfer the methods from xbmc_inputstream_dll.h to XBMC
+   */
+
+  // this are properties given to the addon on create
+  // at this time we have no parameters for the addon
+  typedef struct AddonProps_InputStream /* internal */
+  {
+    int dummy;
+  } AddonProps_InputStream;
+
+  typedef struct AddonToKodiFuncTable_InputStream /* internal */
+  {
+    KODI_HANDLE kodiInstance;
+    DemuxPacket* (*allocate_demux_packet)(void* kodiInstance, int data_size);
+    void (*free_demux_packet)(void* kodiInstance, DemuxPacket* packet);
+  } AddonToKodiFuncTable_InputStream;
+
+  struct AddonInstance_InputStream;
+  typedef struct KodiToAddonFuncTable_InputStream /* internal */
+  {
+    kodi::addon::CInstanceInputStream* addonInstance;
+
+    bool (__cdecl* open)(AddonInstance_InputStream* instance, INPUTSTREAM&);
+    void (__cdecl* close)(AddonInstance_InputStream* instance);
+    const char* (__cdecl* get_path_list)(AddonInstance_InputStream* instance);
+    void (__cdecl* get_capabilities)(AddonInstance_InputStream* instance, INPUTSTREAM_CAPABILITIES*);
+
+    // IDemux
+    struct INPUTSTREAM_IDS (__cdecl* get_stream_ids)(AddonInstance_InputStream* instance);
+    struct INPUTSTREAM_INFO (__cdecl* get_stream)(AddonInstance_InputStream* instance, int);
+    void (__cdecl* enable_stream)(AddonInstance_InputStream* instance, int, bool);
+    void (__cdecl* demux_reset)(AddonInstance_InputStream* instance);
+    void (__cdecl* demux_abort)(AddonInstance_InputStream* instance);
+    void (__cdecl* demux_flush)(AddonInstance_InputStream* instance);
+    DemuxPacket* (__cdecl* demux_read)(AddonInstance_InputStream* instance);
+    bool (__cdecl* demux_seek_time)(AddonInstance_InputStream* instance, double, bool, double*);
+    void (__cdecl* demux_set_speed)(AddonInstance_InputStream* instance, int);
+    void (__cdecl* set_video_resolution)(AddonInstance_InputStream* instance, int, int);
+
+    // IDisplayTime
+    int (__cdecl* get_total_time)(AddonInstance_InputStream* instance);
+    int (__cdecl* get_time)(AddonInstance_InputStream* instance);
+
+    // IPosTime
+    bool (__cdecl* pos_time)(AddonInstance_InputStream* instance, int);
+
+    // Seekable (mandatory)
+    bool (__cdecl* can_pause_stream)(AddonInstance_InputStream* instance);
+    bool (__cdecl* can_seek_stream)(AddonInstance_InputStream* instance);
+
+    int (__cdecl* read_stream)(AddonInstance_InputStream* instance, uint8_t*, unsigned int);
+    int64_t(__cdecl* seek_stream)(AddonInstance_InputStream* instance, int64_t, int);
+    int64_t (__cdecl* position_stream)(AddonInstance_InputStream* instance);
+    int64_t (__cdecl* length_stream)(AddonInstance_InputStream* instance);
+    void (__cdecl* pause_stream)(AddonInstance_InputStream* instance, double);
+    bool (__cdecl* is_real_time_stream)(AddonInstance_InputStream* instance);
```

Comment:
I think you should either supply variable names for all parameters or for none. Currently, you supply a name for AddonInstance_InputStream parameter, but none for all other paramters...

Maybe we should also change this for addon types that came in with older PRs, where we missind this inconsistency during review.